### DISCUSSION TITLE
Why doesn't pwsh pick up my updated modules?

### DISCUSSION THREAD

**Discussion by bobbrow** [PowerShell/PowerShell#16568] [Q&A] [answered]
Newbie here,

I've been trying to debug a script that I didn't create.  It has two pieces, a `ps1` and a `psm1` that is `Import-Module`d from the `ps1`.  I run the `ps1` in a pwsh window, make a change to the `psm1` and then run the `ps1` again, but it still uses the old content from the `psm1` (yes, I saved the file).  Does anyone know why this happens?  It seems unintuitive to me that I have to close the pwsh window and start a new one every time I update the `psm1`.  Surely I'm doing something wrong.

**Comment by iSazonov** [comment, marked as answer]
Modules are loaded in PS session and don't reloaded for performance reasons. You need to explicitly call Import-Module with Force switch to reload the module.
